cmd/cluster.go

```diff
@@ -666,13 +669,38 @@ func clusterKubeconfig(args []string) error {
 		return fmt.Errorf("active user %s has no oidc authProvider, check config", authContext.User)
 	}
 
-	mergedKubeconfig, err := helper.EnrichKubeconfigTpl(kubeconfigContent, authContext)
+	if !viper.GetBool("merge") {
+		mergedKubeconfig, err := helper.EnrichKubeconfigTpl(kubeconfigTpl, authContext)
+		if err != nil {
+			return err
+		}
+
+		fmt.Println(string(mergedKubeconfig))
+		return nil
+	}
+
+	currentCfg, filename, _, err := auth.LoadKubeConfig(kubeconfigFile)
 	if err != nil {
 		return err
 	}
 
-	// print kubeconfig
-	fmt.Println(string(mergedKubeconfig))
+	clusterResp, err := cloud.Cluster.FindCluster(cluster.NewFindClusterParams().WithID(id), nil)
+	if err != nil {
+		return err
+	}
+
+	mergedKubeconfig, err := helper.MergeKubeconfigTpl(kubeconfigTpl, currentCfg, *clusterResp.Payload.Name, authContext)
```

Comment:
what if clustername  clashes with existing context ?

Reply:
It always updates and ... I was also thinking about longer context names like `<tenant>-<project-name>-<cluster-name>` but then it did not look so good. I think when a user decides to merge kubeconfigs together, he can assume that existing entries will be updated.

Reply:
really ?